li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/producer/LiKafkaProducerImpl.java

```diff
@@ -238,6 +242,16 @@ private LiKafkaProducerImpl(LiKafkaProducerConfig configs,
       Integer partition = producerRecord.partition();
       Future<RecordMetadata> future = null;
       UUID messageId = _uuidFactory.getUuid(producerRecord);
+
+      long eventTimestamp = System.currentTimeMillis();
```

Comment:
_auditor below is using timestamp defined in line 241. Should it be used instead?

Reply:
At first, I was under the impression that the timestamps should be completely dictated by the client. This is to avoid end users abusing the header value. Certain downstream systems like auditing may be impacted if timestamps are way off (or invalid). But on second thought, there is a very valid use case in change capture systems who are more interested in the timestamp at which a particular object was updated in the backend store as opposed to the current timestamp. I will just make it equivalent to TmeAuditor's timestamp and any -ve effects that this may have will be handled by the downstream.